src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -447,13 +506,28 @@ await foreach (var streamItem in enumerable)
             }
         }
 
-        private static async Task<object> ExecuteHubMethod(ObjectMethodExecutor methodExecutor, THub hub, object[] arguments)
+        private async Task<HubResult> ExecuteHubMethod(ObjectMethodExecutor methodExecutor, THub hub, object[] arguments, HubConnectionContext connection, IServiceProvider serviceProvider)
+        {
+            if (_invokeMiddleware != null)
+            {
+                var invocationContext = new HubInvocationContext(connection.HubCallerContext, serviceProvider, hub, methodExecutor.MethodInfo, arguments);
+                return await _invokeMiddleware(invocationContext);
+            }
+
+            // If no Hub filters are registered
+            return HubResult.WithResult(await ExecuteMethod(methodExecutor.MethodInfo.Name, hub, arguments));
+        }
+
+        private async Task<object> ExecuteMethod(string hubMethodName, Hub hub, object[] arguments)
         {
+            _methods.TryGetValue(hubMethodName, out var methodDescriptor);
```

Comment:
We shouldn't be paying this cost.

Reply:
> The HubInvocationContext should have an internal ObjectMethodExecutor that use can internally use to invoke methods once we get to this point in the pipeline.

I've been avoiding setting internals on the context because if someone makes a new HubInvocationContext and passes that to `next` the internals won't be there.

Reply:
That’s ok optimize for the 90% case where they don’t

Reply:
What happens if this is null btw?